src/components/Menu/Menu.tsx

```diff
@@ -92,6 +92,11 @@ interface MenuProps {
      */
     searchEmptyText?: string;
 
+    /**
+     * Use this prop when the parent element has position: relative so that the appropriate method of positioning is used
+     */
+    relativelyPositionedParent?: boolean;
```

Comment:
agree that `position?: ['fixed', 'absolute']` will be more clear. 
If we are in absolute, we should check what is the first positioned ancestor - which is not necessarily the anchorEl. usePositioning() has access to the itemRef, so it should not be an issue to look into the ancestors to check which element is the position reference. getPosition could take this "reference" element as a parameter to calculate a valid relative position.
We could do the same if we are in fixed, looking for the first transform, perspective, or filter ancestor, although i don't think it's really necessary, if "absolute" correctly supports all cases.

Reply:
I like the proposed naming with `position` as the prop better.

To avoid having to do the check, why don't we have `position: fixed` be the default and only if `position="absolute"` is passed into the component that we have the menu be absolutely positioned? This implies that the user is consciously choosing to absolutely position the menu and they are setting `position: relative` on the appropriate parent element.

If you don't think the check of parent elements is too expensive, I'm happy to do it that way as well.

I also think that the contextual menu vs usage as a static/dropdown menu are sufficiently different use cases to warrant having two separate components - that could simplify both. The contextual menu would always be `position: fixed` and the static/dropdown menu would always be `position: absolute`.

Reply:
Agree that the user  should be conscious of what he's doing when setting to absolute, but how would you calculate the position and what reference will you use ? would you consider the anchorEl is the position reference (and so do not add the anchorEl position to get the element position , like what you do when relativelyPositionedParent is set ?) 
 
Note that in our case, contextual menu is added in a portal just under a dedicated div in the body, so using fixed will work (as it is right now), using absolute is ok if we keep same as relativelyPositionedParent=false, but would be broken if we use absolute with the behaviour of relativelyPositionedParent=true . The positioned parent is the body, which is different from the anchorEl.

Reply:
Yes, I think the user should use `position: absolute` with the understanding that the `anchorEl` should have `position: relative` and that's how the positioning will get calculated. Is this too much to ask of the user?

To solve for most cases, it seems like keeping `position: fixed` the default will suffice.

Reply:
It's ok for me if "fixed" is the default. It just need to be documented that when using position:absolute , we'll consider that the `anchorEl` is the closesest positioned ancestor.

Reply:
we could also add a new optional prop `positionedAncestorEl` to pass the closest positioned ancestor, if it's not `anchorEl` :-) we would need to change getPositionRelativeToEl to calculate the offset different between anchorEl and `positionedAncestorEl` .. don't know if it would be useful ..

Reply:
After discussion, decided to keep `position: fixed` as the default and keep the exist behavious with `anchorEl.` When `position: absolute` is specified via the new prop, the usePositioning hook will search for the nearest ancestor with `position: relative` and anchor against that.